src/DataLib/fendl32B_retrofit/groupr_tools.py

```diff
@@ -107,4 +108,95 @@ def write_njoy_input_file(template):
+    this method ensures that they are present. Edits will only be made if
+    the SEND record for MF1 or the FEND record for MF3 are not present.
+    The formatting for these records can be found at:
+    https://t2.lanl.gov/nis/endf/intro06.html
+
+    Arguments:
+        gendf_path (str): File path to the newly created GENDF file.
+        matb (int): Unique material ID for the material in the GENDF file.
+    
+    Returns:
+        None
+    """
+
+    # In ENDF-6 formatted files, there are 66 characters/columns of whitespace
+    # before the values in record-keeping lines
+    whitespace = ' ' * 66
+
+    # Define identifiers and corresponding new lines
+    mf1_identifier = f'{matb} 1451   '
+    mf3_identifier = f'{matb} 3  099999'
+    matb = str(matb).zfill(4)
+    mf1_SEND_RECORD = f'{whitespace}{matb} 1  099999'
+    mf3_FEND_RECORD = f'{whitespace}{matb} 0  0    0'
+
+    with open(gendf_path, 'r') as gendf_file:
+        file_str = gendf_file.read()
+    file_lines = file_str.splitlines()
+
+    updates = [
+        (mf3_identifier, mf3_FEND_RECORD),
+        (mf1_identifier, mf1_SEND_RECORD)
+    ]
+
+    for identifier, new_line in updates:
+        last_line_index = file_str.rfind(identifier)
+        line_number = file_str[:last_line_index].count('\n')
+        file_lines.insert(line_number + 1, new_line)
+
+    new_file_str = '\n'.join(file_lines) + '\n'
+
+    with open(gendf_path, 'w') as gendf_file:
+        gendf_file.write(new_file_str)
+
+def run_njoy(element, A, matb):
+    """
+    Use subprocess to run NJOY given a pre-written input card to convert a
+        pair of ENDF and PENDF files to a GENDF file and save it locally.
+        The ENDF and PENDF files must be stored in the same directory and be
+        titled 'tape20' and 'tape21' respectively for a successful NJOY run.
+    
+    Arguments:
+        cards (dict): Dictionary containing each "card"
+            identified by its card number.
+        element (str): Chemical symbol for element of interest.
+        A (str or int): Mass number for selected isotope.
+            If the target is an isomer, "m" after the mass number,
+            so A must be input as a string.
+        matb (int): Unique material ID for the material in the files.
+    
+    Returns:
+        gendf_path (str): File path to the newly created GENDF file. 
+    """
+
+    # Define the input files
+    INPUT = 'groupr.inp'
+    OUTPUT = 'groupr.out'
+
+    # Run NJOY
+    result = subprocess.run(['njoy'], input=open(INPUT).read(),
+                            text=  True, capture_output=True)
+    with open(OUTPUT, 'w') as output_file:
+        output_file.write(result.stdout)
+
+    # If the run is successful, log out the output
+    # and make a copy of the file as a .GENDF file
+    if not result.stderr:
+        output = subprocess.run(['cat', 'output'],
+                                capture_output=True, text = True)
```

Comment:
I don't see where you use this `output` variable? Even if you did, isn't it easier to just read the file?